contrib/scalar/scalar.c

```diff
@@ -695,6 +695,44 @@ static char *remote_default_branch(const char *url)
 	return NULL;
 }
 
+static void strbuf_parentdir(struct strbuf *buf)
+{
+	for (; buf->len > 0 && !is_dir_sep(buf->buf[buf->len]); buf->len--)
+		; /* keep looking for parent directory */
+	buf->buf[buf->len] = '\0';
+}
```

Comment:
It might be more idiomatic to do this:

```suggestion
static void strbuf_parentdir(struct strbuf *buf)
{
	int i;
	for (i = buf->len - 1; i >= 0; i--) {
		if (is_dir_sep(buf->buf[i]))
			break;
	}
	strbuf_setlen(buf, i);
}
```

Reply:
Yeah I shamelessly took this snippet from `setup_enlistment_directory`, should that be changed?

Reply:
Could `setup_enlistment_directory` take advantage of this call?

(Also, we might want to consider putting this in `strbuf.[ch]` eventually.)

Reply:
Updated to match the fixed source from `setup_enlistment_directory`.

> Could setup_enlistment_directory take advantage of this call?

Not sure.. it's also interleaving some other path manipulation between the loop and setlen.